File: 프로그래머스/unrated/181932. 코드 처리하기/코드 처리하기.java
class Solution {
    public String solution(String code) {
        String answer = "";
        int mode = 0;  // Initial mode is 0

        for (int idx = 0; idx < code.length(); idx++) {
            if (mode == 0) {
                if (code.charAt(idx) != '1') {
                    if (idx % 2 == 0) {
                        answer += code.charAt(idx);
                    }
                } else {
                    mode = 1;  // Change mode to 1
                }
            } else {
                if (code.charAt(idx) != '1') {
                    if (idx % 2 != 0) {
                        answer += code.charAt(idx);
                    }
                } else {
                    mode = 0;  // Change mode to 0
                }
            }
        }

        if (answer.isEmpty()) {
            return "EMPTY";
        }

        return answer;
    }
}

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.py
def solution(code):
    answer = ""
    mode = 0  # Initial mode is 0

    for idx, char in enumerate(code):
        if mode == 0 and char != '1' and idx % 2 == 0:
            answer += char
        elif mode == 1 and char != '1' and idx % 2 != 0:
            answer += char
        elif char == '1':
            mode = 1 - mode

    return answer if answer else "EMPTY"
